pkg/controller.v1/mpi/mpijob_controller.go

```diff
@@ -747,11 +747,10 @@ func (jc *MPIJobReconciler) getOrCreateLauncherServiceAccount(mpiJob *kubeflowv1
 		return nil, err
 	}
 	// If the launcher ServiceAccount is not controlled by this MPIJob resource, we
-	// should log a warning to the event recorder and return.
+	// should log a warning to the event recorder.
 	if !metav1.IsControlledBy(sa, mpiJob) {
 		msg := fmt.Sprintf(MessageResourceExists, sa.Name, sa.Kind)
 		jc.Recorder.Event(mpiJob, corev1.EventTypeWarning, ErrResourceExists, msg)
-		return nil, fmt.Errorf(msg)
```

Comment:
Why do we need to remove this return ?

Reply:
If we return with error here, then the workload will not be scheduled. The check in line 751 is made to make sure that the service account was just created and that no existing service account was used. Since we want to use an existing service account, this need to be removed then.

Reply:
Hmm, do we really need to check if that service account is controller by MPIJob if we allow user to create custom service accounts ?
Maybe we could completely remove this check and not return Warning (e.g. remove line 751 - 754) ?
WDYT @tenzen-y @alculquicondor @johnugeorge ?

Reply:
How do we proceed here?

Reply:
I would suggest to remove these lines: https://github.com/kubeflow/training-operator/blob/a193b0b1567a243f785454d61f0614a1ae2a45ad/pkg/controller.v1/mpi/mpijob_controller.go#L749-L754

Reply:
@andreyvelich : Applied your suggestions :)

Reply:
this looks about right, but I'm no longer an expert in mpijob v1

Reply:
I'm ok with this.